File: Backend/src/main/java/com/example/Java_Backend/entities/ProductDetail.java
package com.example.Java_Backend.entities;

import com.example.Java_Backend.entities.Product;
import com.example.Java_Backend.entities.ProductGrindingOption;
import com.example.Java_Backend.entities.GrindingOption;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
public class ProductDetail {
    private Integer Id;

    private Integer ProductId;

    private String Region;

    private String Process;

    private String Roast;

    private String FlavorNotes;

    private String Weight;

    private String Height;

    private Product product;

    private List<GrindingOption> GrindingOption;

}

File: KHE_AuthService/KHE_AuthService/Entities/ProductDetail.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace KHE_AuthService.Entities
{
    public class ProductDetail
    {
        public int Id { get; set; }

        public int ProductId { get; set; }

        public string? Region { get; set; }

     
        public string? Process { get; set; }

        public string? Roast { get; set; }

        public string? FlavorNotes { get; set; }

        public byte? AcidityLevel { get; set; }

        public byte? BitternessLevel { get; set; }

        public byte? BodyLevel { get; set; }

        public string? BestTime { get; set; }

        public string? MatchTags { get; set; }

        public virtual Product Product { get; set; } = null!;
    }
}
